src/main/java/seedu/address/ui/InternshipCard.java

```diff
@@ -80,8 +82,16 @@ public InternshipCard(Internship internship, int displayedIndex) {
      * @param contactNumber phone number of the contact person under the Internship entry
      */
     private void setPocLabel(ContactName contactName, ContactEmail contactEmail, ContactNumber contactNumber) {
-        String stringToSet = "POC: " + contactName.toString() + " | "
-                + contactEmail.toString() + " | " + contactNumber.toString();
+        String contactNameStr = (contactName != null) ? contactName.toString() : "";
+        String contactEmailStr = (contactEmail != null) ? contactEmail.toString() : "";
+        String contactNumberStr = (contactNumber != null) ? contactNumber.toString() : "";
+
+        String stringToSet = "POC: " + String.join(" | ", contactNameStr, contactEmailStr, contactNumberStr);
+
+        if (contactName == null && contactEmail == null && contactNumber == null) {
+            stringToSet = "No POC found";
+        }
+
```

Comment:
Does 2 things;
1 - Does not display anything in the space of a particular optional field in the Internship Card if said optional field has no value
2 - Returns No POC found if any one of the values in setPocLabel is null